Makefile

```diff
@@ -115,6 +115,18 @@ release:
 	$(VERSION_CHECK)
 	./scripts/release.sh build-release "$(VERSION_TAG)" "$(BUILD_SYSTEM)" "$(RELEASE_TAGS)" "$(RELEASE_LDFLAGS)"
 
+release-tag:
+	@$(call print, "Adding release tag.")
+
+	@version_file_path="version.go" && \
+	tag=$$(./scripts/get-git-tag-name.sh $$version_file_path); \
+	exit_status=$$?; \
+	if [ $$exit_status -ne 0 ]; then \
+		echo "Script encountered an error with exit status $$exit_status."; \
+	fi; \
+	echo "Adding git tag: $$tag"; \
+	git tag -s -a "$$tag" -m "Tag generated using command \`make release-tag\`.";
```

Comment:
I think the `-a` is implied by `-s` and `-m`.

Reply:
Also, should we move the "$$tag" to the very end? For a moment I thought it was the value for `-a` which I then had to look up first.

Reply:
> I think the `-a` is implied by `-s` and `-m`.

I don't think that's the case. I think that an annotated tag contains more information than just message and signature. An annotated tag should also include the tagger's name, email, and tagging date. Which shouldn't be included in a non-annotated tag.

> Also, should we move the "$$tag" to the very end?

For clarity, I'll rearrange the command as:
```
git tag -as -m "Tag generated using command `make release-tag`." "$$tag"
```

Let me know if there's more I can do for clarity.